File: 02_Codes/03_API/components/mobile/area.js
const { db } = require('../../firebase')

exports.get = async function (req, res) {
  res.setHeader('Content-Type', 'application/json')

  const vegetable = await db.collection('vegetable').get()

  const f_vegetable = (id) => {
    let v_name = ''

    vegetable.forEach((element) => {
      if (element.id == id) {
        v_name = element.data().v_name
      }
    })

    return v_name
  }

  try {
    const data = []
    await db
      .collection('area')
      .get()
      .then((datat) => {
        datat.docs.forEach((doc) => {
          data.push({
            area_id: doc.id,
            area_name: doc.data().area_name,
            area_lv: doc.data().area_lv,
            area_v_name: f_vegetable(doc.data().area_lv),
            area_long: doc.data().area_long,
            area_wide: doc.data().area_wide,
            area_photo: doc.data().area_photo,
            status: doc.data().status,
            area_detail: doc.data().area_detail,
          })
        })
        res.status(200).json(data)
      })
      .catch((err) => {
        res.status(400).json(err)
      })
    //console.log(data);
  } catch (err) {
    return res.status(500).send(err)
  }
}

File: 02_Codes/02_MoblieApplication_Admin/vegetable_plots/lib/models/area.dart
import 'package:meta/meta.dart';
import 'dart:convert';

List<Area> areaFromJson(String str) =>
    List<Area>.from(json.decode(str).map((x) => Area.fromJson(x)));

String areaToJson(List<Area> data) =>
    json.encode(List<dynamic>.from(data.map((x) => x.toJson())));

class Area {
  Area({
    @required this.areaId,
    @required this.areaName,
    @required this.areaLv,
    @required this.areaVName,
    @required this.areaLong,
    @required this.areaWide,
    @required this.areaPhoto,
    @required this.status,
    @required this.areaDetail,
  });

  final String areaId;
  final String areaName;
  final String areaLv;
  final String areaVName;
  final int areaLong;
  final double areaWide;
  final String areaPhoto;
  final bool status;
  final String areaDetail;

  factory Area.fromJson(Map<String, dynamic> json) => Area(
        areaId: json["area_id"],
        areaName: json["area_name"],
        areaLv: json["area_lv"],
        areaVName: json["area_v_name"],
        areaLong: json["area_long"],
        areaWide: json["area_wide"].toDouble(),
        areaPhoto: json["area_photo"],
        status: json["status"],
        areaDetail: json["area_detail"],
      );

  Map<String, dynamic> toJson() => {
        "area_id": areaId,
        "area_name": areaName,
        "area_lv": areaLv,
        "area_v_name": areaVName,
        "area_long": areaLong,
        "area_wide": areaWide,
        "area_photo": areaPhoto,
        "status": status,
        "area_detail": areaDetail,
      };
}
